File: src/main/java/com/example/the_machine/dto/EventEnvelopeDto.java
package com.example.the_machine.dto;

import com.example.the_machine.domain.EventType;
import com.fasterxml.jackson.databind.JsonNode;
import java.util.UUID;

public record EventEnvelopeDto(
    EventType type,
    UUID threadId,
    UUID runId,
    UUID messageId,
    JsonNode data
) {

}

File: src/main/kotlin/com/example/the_machine/dto/EventEnvelopeDto.kt
package com.example.the_machine.dto

import com.example.the_machine.domain.EventType
import kotlinx.serialization.json.JsonElement
import java.util.*

data class EventEnvelopeDto(
  val type: EventType,
  val threadId: UUID,
  val runId: UUID?,
  val messageId: UUID?,
  val data: JsonElement
)